rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java

```diff
@@ -137,8 +138,9 @@ private CachingRlsLbClient(Builder builder) {
             builder.evictionListener,
             scheduledExecutorService,
             timeProvider);
+    String serverHost = URI.create("//" + helper.getAuthority()).getHost();
```

Comment:
I'm wary of this because it can throw, and because it could mess up encodings. If this throws that panics the channel.

It seems this should use the long URI constructor to pass authority in directly. All the arguments but authority can be null. If the URI has a null host or the constructor throws a URISyntaxException, use original authority?

Reply:
Done.